numpy/core/src/multiarray/datetime_strings.c

```diff
@@ -1609,7 +1609,7 @@ _inc_countmax(int count, int max){
 */
 NPY_NO_EXPORT int
 strftime_from_datastruct(npy_datetimestruct dts, PyObject *timezone_obj,
-                         char *fmt, char *buff, int strsize){
+                         char *fmt, char *buff, const int strsize){
```

Comment:
Marking this const is a little weird, and something I would be more in favor of if we did it everywhere.

Reply:
Windows compilers wouldn’t let me init newfmt with it.  I guess I should just make newfmt a pointer and malloc.

Reply:
Oh, interesting. I'd forgotten about VLAs, because numpy has not had c99 available until very recently.

I'd probably lean towards malloc here - but clearly I need to sit down and review this properly, which I haven't yet had a chance to do.

Reply:
No rush - I need to get it to pass the tests first.  I'm quite hampered by the inability to check if I'm leaking references.